charts/opensearch/templates/extraManifests.yaml

```diff
@@ -1,8 +1,25 @@
-{{ range .Values.extraObjects }}
+{{- /* Define a recursive function to convert structured data to a plain string YAML */ -}}
+{{- define "structuredToPlainString" -}}
+  {{- printf "%v" (toYaml . | trimSuffix "\n") -}}
+{{- end -}}
+
+{{- range $objKey, $objValue := .Values.extraObjects }}
 ---
-{{- if typeIs "string" . }}
-{{- tpl . $ }}
+{{- if and (eq $objValue.kind "Secret") (hasKey $objValue "stringData") }}
+apiVersion: v1
+kind: Secret
```

Comment:
Can this change fail upgrade?

Reply:
@TheAlgo I've been using this code on my deployment and have had happen where the Secret (in my case) needs to be updated and it does not error out. Not sure if that is what you were looking for.